src/tess.py

```diff
@@ -0,0 +1,364 @@
+                dist = np.sqrt(dRA**2 + dDec**2)
+                if dist > radius:
+                    continue
+
+            # check the source magnitude is within the range
+            if source and source.mag is not None and mag_range:
+                mag = df[mag_col]
+                if np.all(np.isnan(mag)):
+                    continue
+                mag_diff = abs(source.mag - np.nanmedian(mag))
+                if mag_diff > mag_range:
+                    continue
+
+            # verify that all data for the same oid
+            # has the same filter
+            if len(list(set(df[filt_col]))) > 1:
+                raise ValueError(
+                    f"Expected all data for the same oid to have the same filter, "
+                    f"but the oid {df['oid'].iloc[0]} had filters {filters}."
+                )
+            oid_dfs.append(new_oid_df)
+
+        # split the data into lightcurves
+        # based on the gap between observations
+        data = pd.concat(oid_dfs)
+        data_sort = data.sort_values(by=[time_col], inplace=False)
+        data_sort.reset_index(drop=True)
+
+        dt = np.diff(mjd_conversion(data_sort[time_col]))
+
+        gaps = np.where(dt > gap)[0]
+        gaps = np.append(gaps, len(data))
+        prev_idx = 0
+
+        dfs = []
+        for idx in gaps:
+            df_gap = data_sort[prev_idx : idx + 1]
+            for filt in filters:
+                df = df_gap[df_gap[filt_col] == filt].reset_index(
+                    drop=True, inplace=False
+                )
+
+                # drop the flagged or NaN values
+                if drop_bad:
+                    bad_idx = (df[flag_col] != 0) | (df[mag_col].isna())
+                    df = df[~bad_idx]
+
+                if len(df) > 0:
+                    dfs.append(df)
+
+            prev_idx = idx + 1
+
+        new_datasets = []
+        keep_columns = [
+            time_col,
+            exp_col,
+            filt_col,
+            mag_col,
+            magerr_col,
+            ra_col,
+            dec_col,
+            flag_col,
+        ]
+        for df in dfs:
+            df = df[keep_columns]
+            if len(df) > 0:
+                new_datasets.append(Lightcurve(data=df, **init_kwargs))
+
+        return new_datasets
+
+    def fetch_data_from_observatory(self, cat_row, verbose=0, **kwargs):
+        """
+        Fetch data from TESS for a given source.
+        Must return a dataframe (or equivalent), even if it is an empty one.
+        This must be implemented by each observatory subclass.
+
+        Parameters
+        ----------
+        cat_row: dict like
+            A row in the catalog for a specific source.
+            In general, this row should contain the following keys:
+            name, ra, dec, mag, filter_name (saying which band the mag is in).
+        kwargs: dict
+            Additional keyword arguments to pass to the fetcher.
```

Comment:
I would prefer to decide what the parameters are and name them explicitly. 
In this case replace `**kwargs` with `cone_radius=0.01` and so on.

Reply:
Should dst_threshold and magdiff_threshold be parameters here, or do they get handled in the Parameters object?